src/Symfony/Component/DependencyInjection/Secrets/JweEntry.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace Symfony\Component\DependencyInjection\Secrets;
+
+use Symfony\Component\DependencyInjection\Exception\RuntimeException;
+
+final class JweEntry
+{
+    //TODO do we need to store header and auth tag separately?
+    private $header;
+    private $encryptedKey;
+    private $iv;
+    private $cipherText;
+    private $authTag;
+
+    public static function decrypt(string $compactedEntry, string &$keyPair)
+    {
+        $entry = new self();
```

Comment:
If this class is only meant to be instantiated by itself, we should make a `private function __construct()`.

Reply:
👍